force-app/main/default/classes/ProgramEngagementService.cls

```diff
@@ -0,0 +1,49 @@
+/*
+ *
+ *  * Copyright (c) 2022, salesforce.com, inc.
+ *  * All rights reserved.
+ *  * SPDX-License-Identifier: BSD-3-Clause
+ *  * For full license text, see the LICENSE file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ *
+ */
+
+public with sharing class ProgramEngagementService {
+    @TestVisible
+    private FieldBucketSelector bucketSelector = new FieldBucketSelector();
+
+    @TestVisible
+    private Set<String> activeOrEnrolledStages {
+        get {
+            if (activeOrEnrolledStages == null) {
+                activeOrEnrolledStages = getActiveOrEnrolledStages();
+            }
+            return activeOrEnrolledStages;
+        }
+        set;
+    }
+
+    public Set<String> getActiveOrEnrolledStages() {
```

Comment:
Should the name be more generic. The customers could bucket other Stages other than active and enrolled

Reply:
It's a good point, and maybe worth checking in with product - my choice here was driven by the ACs ("1. Create buckets for program engagement stage a. Active & Enrolled Bucket"). I changed it from "Active & Enrolled" to "Active or Enrolled" only because the "and" is technically not accurate. But I agree that it seems clunky, and I think I will reach out to Meg.

Reply:
Thank you @ptewson-sfdo